app/assets/javascripts/provider/admin/apiconfig/services/service_discovery.js.coffee

```diff
@@ -0,0 +1,59 @@
+$ ->
+  form_source = $('form#new_service_source')[0]
+  form_scratch = $('form#new_service,form#create_service')[0]
+  form_discover = $('form#service_discovery')[0]
+
+  clear_select_options = (select)->
+    $(select).find('option').not(':first').remove()
+
+  clear_namespaces = ()->
+    clear_select_options form_discover.service_namespace
+
+  clear_service_names = ()->
+    clear_select_options form_discover.service_name
+
+  show_service_form = (source, discover_func, scratch_func)->
+    if source == 'discover'
+      form_scratch.classList.add('is-hidden')
```

Comment:
If using jQuery like https://github.com/3scale/porta/pull/35/files#r221654192
`form_scratch.addClass('is-hidden')`